iOS-Weather-App/AppDelegate.swift

```diff
@@ -10,27 +10,38 @@ import UIKit
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+        
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.makeKeyAndVisible()
+        
+        let weatherController = WeatherViewController()
+        let navigationController = UINavigationController(rootViewController: weatherController)
+        onOpenSearchController(controller: weatherController, navController: navigationController)
+        onOpenCurrentDetailsViewController(controller: weatherController, navController: navigationController)
+        
+        window?.rootViewController = navigationController
+        
         return true
     }
+}
 
-    // MARK: UISceneSession Lifecycle
-
-    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
-        // Called when a new scene session is being created.
-        // Use this method to select a configuration to create the new scene with.
-        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+private extension AppDelegate {
+    func onOpenSearchController(controller: WeatherViewController, navController: UINavigationController) {
```

Comment:
prefix on is usually use at clousers, let's use another name for the functions, like makeOpenVC